kura/examples/org.eclipse.kura.example.wire.bool.multiport.provider/src/main/java/org/eclipse/kura/example/wire/bool/multiport/provider/LogicalComponentOptions.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eurotech and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+
+package org.eclipse.kura.example.wire.bool.multiport.provider;
+
+import java.util.Map;
+
+import org.eclipse.kura.wire.graph.BarrierAggregatorFactory;
+import org.eclipse.kura.wire.graph.CachingAggregatorFactory;
+import org.eclipse.kura.wire.graph.PortAggregatorFactory;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LogicalComponentOptions {
+
+    public enum AllowedOperations {
+        AND,
+        OR,
+        XOR,
+        NOR,
+        NAND,
+        NOT
+    }
+
+    private static final String FIRST_OPERAND_NAME_PROP_NAME = "operand.name.1";
+    private static final String SECOND_OPERAND_NAME_PROP_NAME = "operand.name.2";
+    private static final String RESULT_NAME_PROP_NAME = "result.name";
+    private static final String BARRIER_MODALITY_PROPERTY_KEY = "barrier";
+    private static final String BOOLEAN_OPERATION = "boolean.operation";
+
+    private static final String OPERAND_NAME_DEFAULT = "operand";
+    private static final String RESULT_NAME_DEFAULT = "result";
+    private static final boolean BARRIER_MODALITY_PROPERTY_DEFAULT = false;
+    private static final AllowedOperations BOOLEAN_OPERATION_DEFAULT = AllowedOperations.AND;
+
+    private static final Logger logger = LoggerFactory.getLogger(LogicalComponentOptions.class);
+
+    private String firstOperandName;
```

Comment:
noted. Cannot however put final on the AllowedOperations object, since I will not be able to assign it to a default value in the catch of the constructor.